packages/storage/src/components/ImageGallery/hooks/useImageRefs.js

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import useStorageRef from './useStorageRef';
+
+export default function useImageRefs(props) {
+    const { path, filter, maxResults: maxResultsProp, onError } = props;
+    const storageRef = useStorageRef();
+    const [imageRefs, setImageRefs] = useState([]);
+
+    useEffect(() => {
+        const maxResults = maxResultsProp >= 0 ? maxResultsProp : 1000;
+        let isSubscribed = true;
+        if (Array.isArray(filter)) {
+            const imagesFiltered = filter.map((name) => storageRef.child(`${path}/${name}`));
+            setImageRefs(imagesFiltered.slice(0, maxResults));
+        } else if (typeof filter === 'function') {
+            storageRef
+                .child(path)
+                .listAll()
+                .then(({ items }) => {
```

Comment:
https://firebase.google.com/docs/storage/web/list-files#paginate_list_results the idea use maxResults in the server, not in the client